### PR TITLE
Add delete field to structs

### PR DESCRIPTION
**Pull request by arv**

### CODE REVIEW

**Comment by arv** [comment]
@zcstarr I think you asked for this?

File: js/noms/src/struct.js [hunk 1]
```diff
@@ -184,12 +191,42 @@ export class StructMirror<T: Struct> {
...
+    const type = removeFieldFromType(this.type, idx);
+    return newStructWithValues(type, values);
+  }
+}
+
+function removeFieldFromType(type: Type<StructDesc>, idx: number): Type<StructDesc> {
+  const {desc} = type;
+  const fieldMap = Object.create(null);
+  let i = 0;
+  desc.forEachField((n, t) => {
```
Comment (zcstarr) [inline js/noms/src/struct.js]: nit. should we type lambda functions here ?

**Comment by zcstarr** [comment]
@arv lgtm. with one nit, not sure what the policy is, see comment.  BTY

File: js/noms/src/struct.js (hunk 1 above)
Comment (arv) [inline js/noms/src/struct.js]: Not really. Only declare types on exported bindings and where it makes the code more readable. We hardly ever annotate local variables.

**Pull request merged**